### PR TITLE
pcre2: ftp.pcre.org looks defunct

### PR DESCRIPTION
**Pull request by baloo**
###### Motivation for this change

According to pcre.org github is now the official mirror for releases.

cc @ttuegel

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Mindavi** [comment]
Hash is still ok?

File: pkgs/development/libraries/pcre2/default.nix [hunk 1]
```diff
@@ -7,7 +7,7 @@ stdenv.mkDerivation rec {
   pname = "pcre2";
   version = "10.37";
   src = fetchurl {
-    url = "https://ftp.pcre.org/pub/pcre/${pname}-${version}.tar.bz2";
+    url = "https://github.com/PhilipHazel/${pname}/releases/download/${pname}-${version}/${pname}-${version}.tar.bz2";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/pcre2/default.nix:10]:
```suggestion
    url = "https://github.com/PhilipHazel/pcre2/releases/download/pcre2-${version}/pcre2-${version}.tar.bz2";
```

**Comment by samueldr** [comment]
> Hash is still ok?

It's possible. It's not generated code, but uploaded assets:

 - https://github.com/PhilipHazel/pcre2/releases/tag/pcre2-10.37

Though a simple TOFU check by breaking the hash can help verify.

**Review by SuperSandro2000 — APPROVED** [review]
Linked on https://www.pcre.org/

**Comment by SuperSandro2000** [comment]
> > Hash is still ok?
> 
> It's possible. It's not generated code, but uploaded assets:
> 
>     * [PhilipHazel/pcre2@`pcre2-10.37` (release)](https://github.com/PhilipHazel/pcre2/releases/tag/pcre2-10.37)
> 
> 
> Though a simple TOFU check by breaking the hash can help verify.

Verified locally and it is the same hash.

Waiting for ofborg and then merging.

**Comment by baloo** [comment]
> Hash is still ok?

```
$ nix-build . -A pcre2.src --check
checking outputs of '/nix/store/12lw9q06n2j6h5x8hsamdsh3yjk2i6g9-pcre2-10.37.tar.bz2.drv'...

trying https://github.com/PhilipHazel/pcre2/releases/download/pcre2-10.37/pcre2-10.37.tar.bz2
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   658  100   658    0     0   2043      0 --:--:-- --:--:-- --:--:--  2043
100 1688k  100 1688k    0     0   779k      0  0:00:02  0:00:02 --:--:-- 1745k
/nix/store/an15rwsimzw7ncwkxxn22ahkjh73npjk-pcre2-10.37.tar.bz2
```
as far as I can tell, yes.

**Pull request merged**